File: binance4j-market/src/main/java/com/binance4j/market/dto/OrderBookLimit.java
package com.binance4j.market.dto;

import io.swagger.annotations.ApiModelProperty;

/** The order book result size */
public enum OrderBookLimit {
	/** 5 */
	LIMIT_5("5"), @ApiModelProperty("")
	/** 10 */
	LIMIT_10("10"), @ApiModelProperty("")
	/** 20 */
	LIMIT_20("20"), @ApiModelProperty("")
	/** 50 */
	LIMIT_50("50"), @ApiModelProperty("")
	/** 100 */
	LIMIT_100("100"), @ApiModelProperty("")
	/** 500 */
	LIMIT_500("500"), @ApiModelProperty("")
	/** 1000 */
	LIMIT_1000("1000"), @ApiModelProperty("")
	/** 5000 */
	LIMIT_5000("5000");

	final String value;

	/**
	 * @param value The value
	 */
	private OrderBookLimit(String value) {
		this.value = value;
	}

	@Override
	public String toString() {
		return value;
	}
}

File: connectors/src/main/kotlin/com/binance4j/market/dto/OrderBookLimit.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.market.dto

/** The order book result size  */
enum class OrderBookLimit
/**
 * @property value Value
 */(val value: String) {
	/** 5  */
	LIMIT_5("5"),
	
	/** 10  */
	LIMIT_10("10"),
	
	/** 20  */
	LIMIT_20("20"),
	
	/** 50  */
	LIMIT_50("50"),
	
	/** 100  */
	LIMIT_100("100"),
	
	/** 500  */
	LIMIT_500("500"),
	
	/** 1000  */
	LIMIT_1000("1000"),
	
	/** 5000  */
	LIMIT_5000("5000");
	
	override fun toString(): String = value
}